tests/ops/qutrit/test_qutrit_matrix_ops.py

```diff
@@ -0,0 +1,378 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the qutrit matrix-based operations.
+"""
+import pytest
+import numpy as np
+from scipy.stats import unitary_group
+
+import pennylane as qml
+from pennylane.wires import Wires
+from pennylane.operation import DecompositionUndefinedError
+
+U_thadamard_01 = np.multiply(
+    1 / np.sqrt(2),
+    np.array(
+        [[1, 1, 0], [1, -1, 0], [0, 0, np.sqrt(2)]],
+    ),
+)
+
+U_tswap = np.array(
+    [
+        [1, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 1, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 1, 0, 0],
+        [0, 1, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 1, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 1, 0],
+        [0, 0, 1, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 1, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 1],
+    ],
+    dtype=np.complex128,
+)
+
+# TODO: Add tests for adding controls to `QutritUnitary` once `ControlledQutritUnitary` is implemented
+
+
+class TestQutritUnitary:
+    """Tests for the QutritUnitary class."""
+
+    def test_qutrit_unitary_noninteger_pow(self):
+        """Test QutritUnitary raised to a non-integer power raises an error."""
+        op = qml.QutritUnitary(U_thadamard_01, wires="a")
+
+        with pytest.raises(qml.operation.PowUndefinedError):
+            op.pow(0.123)
+
+    # TODO: Check with Olivia about broadcasted QutritUnitary
+    # def test_qutrit_unitary_noninteger_pow_broadcasted(self):
```

Comment:
Tagging @dwierichs here since I see he is doing work with broadcasting :smile: 

David, what exactly does this broadcasting do, and is there a trick to getting the operations to work with it? I have found the `ndim_params` variable, but simply setting that (and doing a dimensionality check on the input) leads to dimensionality mismatches in the tests. Does some other part of the code need to be adjusted?

Reply:
Hi @glassnotes 
Without having looked through this PR carefully (very nice addition btw :) ) the following changes typically should be sufficient:
1. Add `ndim_params`
2. Make sure that `compute_decomposition`, if there is any, defers parameters in such a way that a leading dimension does not "get in the way", i.e. instead of `[qml.RX(x[0], wires=0), qml.RY(x[1], wires=1)]` do `[qml.RX(x[...,0], wires=0), qml.RY(x[...,1], wires=1)]` etc.
3. Make sure the operations that `compute_decomposition` returns also support broadcasting.
4. Write `compute_matrix` in such a way that it is compatible with receiving parameters that have an additional dimension compared to `ndim_params`. That is, `RX.compute_matrix` for example is able to take in a 1D tensor of parameters, and creates a 3-dimensional operation tensor, with the broadcasting dimension being the first. The straightforward way of doing this is to check `qml.math.ndim(param)` and do an if-else-loop.
5. Note that one also can just check `qml.math.ndim(param)`, compare with `ndim_params` and raise an Error if broadcasting attempted. Later we can add the functionality for broadcasting in a separate PR. (This error should optimally be raised both in `__init__` and in `compute_matrix`/`compute_decomposition`, I suppose).

Reply:
Thanks for the insight! That's really helpful. I added the broadcasting related checks and tests for it as well.

Reply:
Thanks so much @dwierichs ! :tada: